build.gradle

```diff
@@ -12,7 +12,24 @@ buildscript {
 }
 
 group 'com.dynatrace.openkit'
-version '${version}'
+
+def releaseVersion
+if(!project.hasProperty('currentVersion')) {
+    Properties properties = new Properties()

```

Comment:
Please move this code into a function, i.e.
```
def readVersion() {
    ...
    return properties.version
}
```
Then set the version directly, if it hasn't been set from the outside (-Pversion=...).
```
if (version == 'unspecified') {
    version readVersion(project)
}
```
This is probably a little more elegant :)